app/controllers/api/mentor_discussion_posts_controller.rb

```diff
@@ -2,10 +2,18 @@ module API
   class MentorDiscussionPostsController < BaseController
     before_action :use_mentor_discussion
 
+    def index
+      posts = @discussion.posts.where(iteration: params[:iteration_id])
+
+      render json: SerializeMentorDiscussionPosts.(posts)
+    end
+
     def create
+      iteration = @discussion.solution.iterations.find_by!(idx: params[:iteration_idx])
```

Comment:
The API should always return proper errors for 404s so you need to copy the [`render_404`](https://github.com/exercism/v3-website/pull/384/files#diff-7cc785777039545fff75c178b122b9a389858f2658009f184cbc81d63b738cb8R38) line below underneath here with an `:iteration_not_found` key. And you need to add a controller test for this situation